pkgs/development/python-modules/nf-core/default.nix

```
@@ -0,0 +1,88 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  setuptools,
+  wheel,
+  click,
+  filetype,
+  gitpython,
+  jinja2,
+  jsonschema,
+  markdown,
+  packaging,
+  pdiff,
+  pillow,
+  pre-commit,
+  prompt-toolkit,
+  pytest,
+  pytest-workflow,
+  pyyaml,
+  questionary,
+  # NOTE Not going to support as it requires 10+ packages and is a niche feature
+  # refgenie
+  requests,
+  requests-cache,
+  rich,
+  rich-click,
+  tabulate,
+  trogon,
+}:
+
+buildPythonPackage rec {
+  pname = "nf-core";
+  version = "2.12.1";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "nf-core";
+    repo = "tools";
+    rev = version;
+    hash = "sha256-LuJm6/v/RFQ999U0LznzsMp3Hnxi3r5cl2iOZ2BtnKE=";
+  };
+
+  nativeBuildInputs = [
+    setuptools
+    wheel
+  ];
```

```suggestion
  build-system = [ setuptools ];
```
Please use `build-system` instead.
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-parameters-buildpythonpackage-parameters

Also, since `wheel` is not a `build-system.requires`, we don't add it to `build-system`.
https://setuptools.pypa.io/en/latest/build_meta.html#how-to-use-it